x-pack/test/api_integration/apis/entity_manager/builtin_definitions.ts

```diff
@@ -0,0 +1,192 @@
+        createAdmin({ esClient }),
+        createRuntimeUser({ esClient }),
+      ]);
+    });
+
+    describe('enablement/disablement', () => {
+      describe('with authorized user', () => {
+        it('should enable and disable entity discovery', async () => {
+          const enableResponse = await enableEntityDiscovery(authorizedUser, 200);
+          expect(enableResponse.success).to.eql(true, "authorized user can't enable EEM");
+
+          const definitionsResponse = await getInstalledDefinitions(
+            supertestWithoutAuth,
+            authorizedUser
+          );
+          expect(definitionsResponse.definitions.length).to.eql(builtInDefinitions.length);
+          expect(
+            builtInDefinitions.every((builtin) =>
+              isInstalledAndRunning(builtin, definitionsResponse.definitions)
+            )
+          ).to.eql(true, 'all builtin definitions are not installed/running');
+
+          let stateResponse = await entityDiscoveryState(authorizedUser, 200);
+          expect(stateResponse.enabled).to.eql(
+            true,
+            `EEM is not enabled; response: ${JSON.stringify(stateResponse)}`
+          );
+
+          const disableResponse = await disableEntityDiscovery(authorizedUser, 200, {
+            deleteData: false,
+          });
+          expect(disableResponse.success).to.eql(
+            true,
+            `authorized user failed to disable EEM; response: ${JSON.stringify(disableResponse)}`
+          );
+
+          stateResponse = await entityDiscoveryState(authorizedUser, 200);
+          expect(stateResponse.enabled).to.eql(false, 'EEM is not disabled');
+
+          await expectNoInstalledDefinitions();
+        });
+      });
+
+      describe('with unauthorized user', () => {
+        it('should fail to enable entity discovery', async () => {
+          await enableEntityDiscovery(unauthorizedUser, 403);
+
+          const stateResponse = await entityDiscoveryState(unauthorizedUser, 200);
+          expect(stateResponse.enabled).to.eql(false, 'EEM is enabled');
+
+          await expectNoInstalledDefinitions();
+        });
+
+        it('should fail to disable entity discovery', async () => {
+          const enableResponse = await enableEntityDiscovery(authorizedUser, 200);
+          expect(enableResponse.success).to.eql(true, "authorized user can't enable EEM");
+
+          await disableEntityDiscovery(unauthorizedUser, 403);
+
+          const disableResponse = await disableEntityDiscovery(authorizedUser, 200);
+          expect(disableResponse.success).to.eql(true, "authorized user can't disable EEM");
+        });
+      });
+    });
+
+    describe('upgrade', () => {
+      it('should noop when no api key stored', async () => {
+        const result = await upgradeBuiltinDefinitions(supertest, []);
+        expect(result).to.eql({ success: false, reason: ERROR_API_KEY_NOT_FOUND });
+      });
+
+      it('should upgrade existing definitions', async () => {
```

Comment:
I wonder if we should test that the update is also reflected in the resulting documents, to catch that the ingest pipelines are updated/don't have any hard coded values but it might not be worth it.

Reply:
Definitely, this will be a follow up once https://github.com/elastic/kibana/pull/190097 is merged